ruby_event_store-rom/lib/ruby_event_store/rom/repositories/events.rb

```diff
@@ -78,8 +64,33 @@ def read(specification)
           end
         end
 
+        def count(specification)
+          query = read_scope(specification)
+          limit = specification.limit if specification.limit?
+          query = query_builder(query, limit: limit)
```

Comment:
@mpraglowski I would say just do `query = query.take(limit) if specification.limit?` and  skip `query_builder` because it does some combining of association data and mapping that you don't need to get the `count`